src/api/MainSolver.cc

```diff
@@ -231,9 +234,15 @@ std::unique_ptr<UnsatCore> MainSolver::getUnsatCore() const {
     if (not config.produce_unsat_cores()) { throw ApiException("Producing unsat cores is not enabled"); }
     if (status != s_False) { throw ApiException("Unsat core cannot be extracted if solver is not in UNSAT state"); }
 
-    UnsatCoreBuilder unsatCoreBuilder{&config, &smt_solver->getResolutionProof(), &pmanager};
+    auto & proof = smt_solver->getResolutionProof();
+    auto unsatCoreBuilder = [&]() -> std::variant<UnsatCoreBuilder, MinUnsatCoreBuilder> {
+        if (not config.minimal_unsat_cores()) { return UnsatCoreBuilder{config, proof, pmanager, termNames}; }
+
+        auto auxSMTSolver = std::make_unique<MainSolver>(logic, config, "unsat core solver");
```

Comment:
It might be more efficient to give this new solver a fresh config. Specifically, the new solver does not need to compute unsat cores anymore.

Reply:
That is a good point.

Reply:
Based on this, I actually reconsidered the design and I think both builder classes can be merged. Since we use `MainSolver` and not `SimpSMTSolver`, it is sufficient to extend `UnsatCoreBuilder` constructor of just `Logic &` parameter, which is fine. The internal SMT solver can be constructed inside only if minimization is desired. Do you agree?

Reply:
That would be very good, I think!

Reply:
Done.